flytekit/types/file/file.py

```diff
@@ -468,7 +469,7 @@ def to_literal(
                 remote_path = ctx.file_access.put_data(source_path, remote_path, is_multipart=False, **headers)
             else:
                 remote_path = ctx.file_access.put_raw_data(source_path, **headers)
-            return Literal(scalar=Scalar(blob=Blob(metadata=meta, uri=remote_path)))
+            return Literal(scalar=Scalar(blob=Blob(metadata=meta, uri=unquote(str(remote_path)))))
```

Comment:
Thanks, could you add a test for it?

Reply:
Done!
(link to [commit](https://github.com/flyteorg/flytekit/pull/2478/commits/06662deede5061b8b3edbcbe1fdec38e3154d4a9))